### PR TITLE
Add required go-version option to generate telemetryInfo without holes

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/generator/generator.ts [hunk 1]
```diff
@@ -41,7 +41,8 @@ export async function protocolGen(host: Host) {
       host.WriteFile(`${filePrefix}${op.name.toLowerCase()}_client.go`, op.content, undefined, 'source-file-go');
     }
 
-    const constants = await generateConstants(session);
+    const version = await session.getValue("go-version", "<version>");
```
Comment (RickWinter) [inline src/generator/generator.ts:44]: Why not make the version parameter mandatory?  I don't a reason not to and it avoids us sending meaningless data of "<version>"

**Review by RickWinter — APPROVED** [review]

File: src/generator/generator.ts (hunk 1 above)
Comment (chamons) [reply src/generator/generator.ts:44]: The current workflow for management folks is to generate code and then modify it by hand. They requested this flag to improve that.

I wasn't sure if we wanted a "breaking" change like this for regeneration? Suggestions on who to ask if this is better?

Comment (RickWinter) [reply src/generator/generator.ts:44]: Are you referring to breaking in process or code?  I'll speak to both,
The telemetry string we send should reflect the version of the library that is running.  Thus it should rev which each revision of the library.  It's not considered breaking to change that string.

For process we should evolve that to where it should be (shift-left).  If that means we need to change (i.e. break process) to enforce constraints and increase the quality of generated code then we should do so.

Comment (chamons) [reply src/generator/generator.ts:44]: I was only concerned on breaking process. 

Sounds like we should make this mandatory.

File: .scripts/regeneration.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ function generate(name, inputFile, outputDir, additionalArgs) {
     sem.take(function() {
         console.log('generating ' + inputFile);
         cleanGeneratedFiles(outputDir);
-        exec('autorest --use=. --file-prefix="zz_generated_" --modelerfour.lenient-model-deduplication --license-header=MICROSOFT_MIT_NO_VERSION --input-file=' + inputFile + ' --output-folder=' + outputDir + ' ' + additionalArgs + ' ' + switches.join(' '), autorestCallback(outputDir, inputFile));
+        exec('autorest --use=. --file-prefix="zz_generated_" --modelerfour.lenient-model-deduplication --license-header=MICROSOFT_MIT_NO_VERSION --go-version=0.1 --input-file=' + inputFile + ' --output-folder=' + outputDir + ' ' + additionalArgs + ' ' + switches.join(' '), autorestCallback(outputDir, inputFile));
```
Comment (ArcturusZhang) [inline .scripts/regeneration.js:131]:
```suggestion
        exec('autorest --use=. --file-prefix="zz_generated_" --modelerfour.lenient-model-deduplication --license-header=MICROSOFT_MIT_NO_VERSION --go-version=v0.1.0 --input-file=' + inputFile + ' --output-folder=' + outputDir + ' ' + additionalArgs + ' ' + switches.join(' '), autorestCallback(outputDir, inputFile));
```

**Review by ArcturusZhang — COMMENTED** [review]
Thank you for the change!
In the examples, we might need to keep the version number in the semantic version way.
And BTW, by this implementation, we will need to assign the version in this way
```
--go-version=v0.1.0
```
with a `v` in front of it, right? Because the `v` prefix is mandatory as a part of the version number.

File: .scripts/regeneration.js (hunk 2 above)
Comment (chamons) [reply .scripts/regeneration.js:131]: Do we want to do this, or should we prefix v in front always?

**Comment by chamons** [comment]
Hmm, I ran `rush regenerate` and don't see other files to change.

File: .scripts/regeneration.js (hunk 2 above)
Comment (chamons) [reply .scripts/regeneration.js:131]: Decided to bundle in the v always.

File: .scripts/regeneration.js [hunk 3]
```diff
@@ -82,7 +82,7 @@ for (namespace in goMappings) {
     // for each swagger run the autorest command to generate code based on the swagger for the relevant namespace and output to the /generated directory
     const entry = goMappings[namespace];
     const inputFile = swaggerDir + entry[0];
-    let extraParams = ['--export-clients'];
+    let extraParams = ['--export-clients', '--go-version=0.1'];
```
Comment (jhendrixMSFT) [inline .scripts/regeneration.js:85]: Per discussion, let's call this `module-version`.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**